ios/FluentUI/Vnext/List/List.swift

```diff
@@ -53,66 +55,64 @@ public struct MSFListView: View {
+                            Header(title: sectionTitle, tokens: tokens)
+                        }
 
-                ForEach(section.cells, id: \.self) { cell in
-                    MSFListCellView(cell: cell,
-                                    tokens: tokens)
-                        .border(section.hasBorder ? Color(tokens.borderColor) : Color.clear, width: section.hasBorder ? tokens.borderSize : 0)
-                        .frame(maxWidth: .infinity)
+                        ForEach(section.cells.indices, id: \.self) { index in
+                            let cell = section.cells[index]
+                            let hasDividers = (index < section.cells.count - 1 && section.hasDividers) || (section.hasDividers && (cell.children != nil))
+                            MSFListCellView(cell: cell,
+                                            tokens: tokens,
+                                            hasDividers: hasDividers)
+                                .frame(maxWidth: .infinity)
+                        }
+                        Divider()
+                            .background(Color(tokens.borderColor))
+                    }
                 }
             }
-        }
-        .environment(\.defaultMinListRowHeight, 0)
-        .frame(maxWidth: .infinity, maxHeight: .infinity)
-        .onAppear {
-            // When environment values are available through the view hierarchy:
-            //  - If we get a non-default theme through the environment values,
-            //    we use to override the theme from this view and its hierarchy.
-            //  - Otherwise we just refresh the tokens to reflect the theme
-            //    associated with the window that this View belongs to.
-            if theme == ThemeKey.defaultValue {
-                self.tokens.updateForCurrentTheme()
-            } else {
-                self.tokens.theme = theme
+            .environment(\.defaultMinListRowHeight, 0)
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .onAppear {
+                // When environment values are available through the view hierarchy:
+                //  - If we get a non-default theme through the environment values,
+                //    we use to override the theme from this view and its hierarchy.
+                //  - Otherwise we just refresh the tokens to reflect the theme
+                //    associated with the window that this View belongs to.
+                if theme == ThemeKey.defaultValue {
+                    self.tokens.updateForCurrentTheme()
+                } else {
+                    self.tokens.theme = theme
+                }
             }
-        }
     }
 }
 
 extension MSFListView {
     /// View for List Cells
     struct MSFListCellView: View {
-        var cell: MSFListCellData
+        var cell: MSFListCellState
         @ObservedObject var tokens: MSFListTokens
+        var hasDividers: Bool
```

Comment:
I think this should be part of the cellState, shouldn't it?

Reply:
I think this is a section property because it should automatically applied per section (on top of including the section dividers). Thoughts?

Reply:
I agree that it's very likely that the divider will be set on the section level rather than on the cell level.
But I think it'd be nice to have the level of granularity on the cell.
We can defer this decision and check with design whether we'll have that kind of scenario (specific cells not having dividers while the section is set otherwise).